end_app/weather_app/views.py

```diff
@@ -49,3 +49,33 @@ def weather_and_forecast_view(request):
         })
 
     return render(request, "weather_and_forecast.html", context)
+
+
+@handle_http_errors
+def air_pollution_view(request):
+    city_form = CityFormForTwoCities(request.POST or None)
+    context = {'city_form': city_form}
+    if request.method == 'POST' and city_form.is_valid():
```

Comment:
sprawdzaj czy form jest walid już w ifie, bo śednio użyć `request.POST` zanim wiesz jaki to request